src/app/layout/qgslayoutlegendwidget.cpp

```diff
@@ -1030,6 +1031,29 @@ void QgsLayoutLegendWidget::mLayerExpressionButton_clicked()
   if ( expressiondialog.exec() )
     layerNode->setLabelExpression( expressiondialog.expressionText() );
 
+  QgsExpressionContextScope *symbolLegendScope = new QgsExpressionContextScope( tr( "Symbol scope" ) );
+
+  QgsFeatureRenderer *r = vl->renderer();
+
+  if ( r )
+  {
+    const QgsLegendSymbolList legendSymbols = r->legendSymbolItems();
+
+    if ( !legendSymbols.empty() )
+    {
+      QgsSymbolLegendNode legendNode( layerNode, legendSymbols.first() );
+
+      symbolLegendScope->addVariable( QgsExpressionContextScope::StaticVariable( QStringLiteral( "symbol_label" ), symbolLabel().remove( "[%" ).remove( "%]" ), true ) );
```

Comment:
@roya0045 could you come up with the code required to reimplement the `symbolLayer()` logic in here?

Reply:
You mean `symbolLabel()` or `symbolLayer()`?

Reply:
simply do `legendNode.symbolLabel().remove( "[%" ).remove( "%]" )`
The function is a public member of the `QgsSymbolLegendNode` see [this line](https://github.com/roya0045/QGIS_1/blob/19d37b4dc95fee889d83d3d114c1d6cdc621560b/src/core/layertree/qgslayertreemodellegendnode.h#L336)

Reply:
Thanks.